server/src/services/storage.service.ts

```diff
@@ -63,6 +68,38 @@ export class StorageService extends BaseService {
   }
 
   @OnJob({ name: JobName.DELETE_FILES, queue: QueueName.BACKGROUND_TASK })
+  async createReadmeFile(folder: StorageFolder) {
+    const { folderPath } = this.getMountFilePaths(folder);
+    const readmePath = join(folderPath, 'README.TXT');
+    const warningMessage =
+      `WARNING: Do not modify/delete the contents of the Upload folder as it may cause unexpected errors and database incompatibilities.\n\n` +
+      `Please refer to the documentation for more information:\n` +
+      `https://immich.app/docs/administration/backup-and-restore#asset-types-and-storage-locations`;
+
+    try {
+      let exists = false;
+      try {
+        await this.storageRepository.stat(readmePath);
+        exists = true;
+      } catch (error) {
```

Comment:
Nested exceptions are imo always ugly and avoidable in almost all cases. Could you refactor this?